### PR TITLE
Add a shell script to register with standard CLI tools

### PR DESCRIPTION
**Pull request by jaeaster**
# Description
An example script using `cast` to register for the Lagrange ZK Coprocessor AVS.
This is an alternative to our Rust CLI. Operators can modify it to suit your operational setup.

This script does 3 things:
1. Generate a new `LAGR_KEY` to be used to authenticate with the Gateway + sign the proofs operators generate. It is needed at runtime while running the worker.
2. Sign a message with your Eigenlayer `ETH_KEY`. This is the very sensitive [withdrawal key](https://docs.eigenlayer.xyz/eigenlayer/operator-guides/key-management/intro#withdrawal-keys)
3. Send a transaction to the `ZKMRStakeRegistry` to register for the ZK Coprocessor AVS

The transaction requires as input:
  1. The public key component of the newly generated `LAGR_KEY`
  2. A signed message from the `ETH_KEY`

It must be signed with the `ETH_KEY` i.e. `cast send --private-key $ETH_KEY`

Operators should update the "MODIFY ME!" sections to load their `ETH_KEY` safely.
They don't need to load the ETH_KEY into the environment at all if they use a remote signer.
They should just modify the `cast send` and `cast sign` commands to sign the hash + tx in a way that meets their security needs.

They also need to update the `OPERATOR_ADDR` variable to match their address.

# Dependencies
1. zsh
2. `cast` - a binary from the foundry toolkit https://github.com/foundry-rs/foundry?tab=readme-ov-file
3. `openssl` - to generate your new ECDSA secp256k1 LAGR_KEY to be used only for the ZK Coprocessor AVS
4. `jq` - only needed for the demo; see the first `=== MODIFY ME! ===` section

### CODE REVIEW

File: register-with-shell.sh [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+store_lagrange_key() {
+    # export LAGR_KEY
+    # echo "LAGR_KEY: $LAGR_KEY"
+    echo "Implement the \`store_lagrange_key\` function before registering"
+    exit 1
+}
+
+# Determine contract addresses based on the chain ID
+set_contract_addresses() {
+    local chain_id = $(cast chain-id)
```
Comment (ChainDev931105) [inline register-with-shell.sh]:
```suggestion
    local chain_id=$(cast chain-id)
```

It returns this error.
```
line 69: local: `=': not a valid identifier
line 69: local: `17000': not a valid identifier
```

**Review by ChainDev931105 — CHANGES_REQUESTED** [review]
dropped one note

File: register-with-shell.sh (hunk 1 above)
Comment (jaeaster) [reply register-with-shell.sh]: Great catch. A typo after the refactor. Fixed now. Thank you!

**Review by ChainDev931105 — APPROVED** [review]

File: register-with-shell.sh [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        echo "Error: Unable to calculate registration hash."
+        exit 1
+    fi
+    printf "\nRegistration hash:\n${HASH}\n"
+}
+
+# Sign the registration hash with your ETH_KEY
+sign_registration_hash() {
+    SIGNATURE=$(cast wallet sign --private-key $ETH_KEY $hash)
+    printf "\nRegistration signature:\n${SIGNATURE}"
```
Comment (ChainDev931105) [inline register-with-shell.sh]:
```suggestion
    SIGNATURE=$(cast wallet sign --private-key $ETH_KEY $HASH)
    printf "\nRegistration signature:${SIGNATURE}\n"
```

**Review by ChainDev931105 — CHANGES_REQUESTED** [review]

File: register-with-shell.sh (hunk 2 above)
Comment (jaeaster) [reply register-with-shell.sh]: fixed

**Review by ChainDev931105 — APPROVED** [review]

**Pull request merged**